gnosis/verification.k

```diff
@@ -109,6 +109,16 @@ module VERIFICATION
     rule store(store(M, K0, V0), K1, V1) => store(store(M, K1, V1), K0, V0)
       requires K0 =/=Int K1 andBool K1 in keys(M)
 
+    syntax IMap ::= "#unusedIMap" [function, smtlib(unusedIMap) ]
+
+    //Hack to delete the BuiltinMap from innermost position and help matching [encodeTransactionData-internal]
+    //Required because BuiltinMap is not translatable to Z3.
+    rule storeRange( storeRange( storeRange( storeRange( M:Map => #unusedIMap,
+                              64,                   32,       #buf(32, INIT_USED_MEM_PTR) ),
+                              DATA_LEN_LOC,         32,       #buf(32, DATA_LEN) ),
+                              DATA_BUF_LOC,         DATA_LEN, #buf(DATA_LEN, DATA) ),
+                              INIT_USED_MEM_ACTUAL, 32,       _ )
+
```

Comment:
i'm not positive with this.  what if the rule is applied in `execTransaction` (i.e., the caller of `encodeTransactionData`) that may read `M`?

Reply:
also, i've not been convinced yet that this rule is necessary.  i mean, i guess that this can be avoided by using the `matching` attribute (possibly with the combination of `smtlib`). no?

Reply:
It was necessary when localMem was in `storeRange` format. Now it's no longer necessary. Deleted.